### ISSUE TITLE
Disable prefer-named-capture-group in String.replace()

### ISSUE THREAD

**Issue by regseb** [ota-meshi/eslint-plugin-regexp#336]
Hi,

Named capture groups aren't usable in RegExp passed to [`String.prototype.replace()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) and [`String.prototype.replaceAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) methods.

The [prefer-named-capture-group](https://ota-meshi.github.io/eslint-plugin-regexp/rules/prefer-named-capture-group.html) rule could have an option to disable this rule in the `String.prototype.replace()` and `String.prototype.replaceAll()` methods.

In the following example, _Capture group '(.*?)' should be converted to a named or non-capturing group (regexp/prefer-named-capture-group)_ are reported.

```JavaScript
const strip = function (text) {
    return text.replaceAll(/\[B\](.*?)\[\/B\]/gu, "$1")
               .replaceAll(/\[I\](.*?)\[\/I\]/gu, "$1")
               .replaceAll(/\[LIGHT\](.*?)\[\/LIGHT\]/gu, "$1")
               .replaceAll(/\[COLOR [^\]]+\](.*?)\[\/COLOR\]/gu, "$1")
               .replaceAll(/\[UPPERCASE\](.*?)\[\/UPPERCASE\]/gu, "$1")
               .replaceAll(/\[LOWERCASE\](.*?)\[\/LOWERCASE\]/gu, "$1")
               .replaceAll(/\[CAPITALIZE\](.*?)\[\/CAPITALIZE\]/gu, "$1")
               .replaceAll("[CR]", " ")
               .trim();
};
```

Maybe there is a better practice to remove tags ?

**Comment by ota-meshi** [comment]
What does it mean to "Named capture groups aren't usable"?

I think you can use a named capture group.

```js
"[B]Foo[/B] [B]Foo[/B]".replaceAll(/\[B\](?<foo>.*?)\[\/B\]/gu, "$<foo>")
// 'Foo Foo'
```

**Comment by regseb** [comment]
I didn't know you could use named capture groups in the replacement string of `replace()`. I thought only `"$0"`, `"$1"`, ... worked. Thanks for the tip.

**Comment by RunDevelopment** [comment]
Btw. you don't have to change the replacement string. You can keep using `$1`.

```js
> "aa".replace(/(a)/, "$1b")
'aba'
> "aa".replace(/(?<foo>a)/, "$1b")
'aba'
```
